### PR TITLE
Carousel

### PR DESCRIPTION
**Pull request by joerghebestreit**
carousel added

### CODE REVIEW

File: client/src/components/Carousel.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    const nextSlide = () => {
+        setPicture(picture === length - 1 ? 0 : picture + 1)
+    }
+
+    const prevSlide = () => {
+        setPicture(picture === 0 ? length -1 : picture - 1)
+    }
+
+    console.log(picture)
```
Comment (FriederikeM) [inline client/src/components/Carousel.js]: You can delete the console log :)

File: client/src/components/Carousel.css [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    box-shadow: 0px 0px 5px 0px rgb(85, 85, 85);
+}
+
+.right-arrow{
+    position: absolute;
+    top: 48%;
+    right: 22px;
+    font-size: 1.7rem;
+    color: white;
+    user-select: none;
```
Comment (FriederikeM) [inline client/src/components/Carousel.css:22]: very interesting, never seen this css property :)

File: client/src/components/Details.js [hunk 3]
```diff
@@ -1,16 +1,13 @@
...
-}
-
 export default function Details() {
   const { ID } = useParams();
   const history = useHistory();
-  const [property, setProperty] = useState({});
+  const [property, setProperty] = useState();
  
     console.log("Hallo von Details")
     console.log(ID)
```
Comment (FriederikeM) [inline client/src/components/Details.js]: you can also delete these :)

**Review by FriederikeM — APPROVED** [review]
Hey Jörg, this looks great. To me it looks like pure magic :D 
Unfortunately the vercel deployment gives me an error, so I can't see what your carousel looks like. Would love to see it 👀

File: client/src/components/Carousel.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import "../components/Carousel.css"
+import {FaArrowAltCircleRight, FaArrowAltCircleLeft} from "react-icons/fa";
+import {useState} from "react";
+
+export default function Carousel({images}) {
+    const [picture, setPicture] = useState(0)
+    const length = images.length
+
+    const nextSlide = () => {
+        setPicture(picture === length - 1 ? 0 : picture + 1)
```
Comment (marwinburesch) [inline client/src/components/Carousel.js]: You could just replace this with:
```suggestion
        setPicture((picture +1) % images.length )
```
and similar but a bit different for the `previousSlide` function. Check out this article about how it works: https://dev.to/ranewallin/this-simple-math-hack-lets-you-create-an-image-carousel-without-any-if-statements-5chj

File: client/src/components/Details.js [hunk 5]
```diff
@@ -25,39 +22,44 @@ export default function Details() {
...
+              <p className="p-m1">Bedrooms</p>
+              <p className="p-m1">Status</p>
+            </div>
+            <div className="p-2-2">
+              <p className="p-m2">{property.YearBuild}</p>
+              <p className="p-m2">{property.Interior}</p>
+              <p className="p-m2">{property.Bedrooms}</p>
+              <p className="p-m2">{property.Status}</p>
+            </div>
+          </div>
```
Comment (marwinburesch) [inline client/src/components/Details.js]: I see some potential for semantic elements like `<section>` for those groups, for example.

I still don't like the class names and what they represent... I think I can remember talking about this in an earlier review. They seem to have effects on the padding property, just judging by the name. However, they don't. Take a look at [the bootstrap docs](https://getbootstrap.com/docs/5.0/utilities/spacing/) for example

**Review by marwinburesch — CHANGES_REQUESTED** [review]
Good work!
Just some comments :)

File: client/src/components/Carousel.css (hunk 2 above)
Comment (joerghebestreit) [reply client/src/components/Carousel.css:22]: its on an imported icon. It won't be selected when you accidentally double click or mark. so it don't behaves like a text. as far as I understood this :)

File: client/src/components/Carousel.js (hunk 1 above)
Comment (joerghebestreit) [reply client/src/components/Carousel.js]: ju, thanks

**Pull request merged**